docs/markdown/Reference-manual.md

```diff
@@ -311,7 +311,8 @@ following.
 
 - `build_by_default` *(added 0.38)* causes, when set to true, to
   have this target be built by default, that is, when invoking plain
-  `ninja`; the default value is false
+  `ninja` or `ninja test` or anything that depends on the default target.
+   The default value is false
```

Comment:
This sentence is pretty hard to understand! How about:

```
- `build_by_default` *(added 0.38)* when set to `true`, this target will be built by default; 
  which means it will be automatically built when `ninja` is called without any arguments
  or when a target that depends on the [default target](https://ninja-build.org/manual.html#_default_target_statements) 
  `all` is specified, such as `ninja test`. The same behavior applies for other backends. 
  The default value is `false`.
```

Reply:
Thanks @nirbheek for the feedback. I was trying to keep it as short as possible but if that's not a major concern then your text looks great to me. Considering you basically wrote it all would you like to submit this yourself in a new PR so `git blame` reflects reality? This would also save me taking credit for the "applies to other backends" bit that I know absolutely nothing about :-) If you're too busy and prefer me to do it that's fine too, up to you.

Reply:
This describes the behaviour with ninja, then says "the same behavour applies for other backends", so perhaps the phrasing could be simplified to not mention the backend at all?

Reply:
I like sample commands like `ninja test` because they get the point across clearly and briefly. This being said, maybe @nirbheek 's text can be rephrased and mention ninja fewer times.